awx/main/tasks.py

```diff
@@ -1146,6 +1149,44 @@ def run(self, pk, **kwargs):
         '''
         Run the job/task and capture its output.
         '''
+        self.instance = self.model.objects.get(pk=pk)
+        containerized = self.instance.is_containerized
+        pod_manager = None
+        if containerized:
+            # Here we are trying to launch a pod before transitioning the job into a running
+            # state. For some scenarios (like waiting for resources to become available) we do this
+            # rather than marking the job as error or failed. This is not always desirable. Cases
+            # such as invalid authentication should surface as an error.
+            from awx.main.scheduler.kubernetes import PodManager # Avoid circular import
+            pod_manager = PodManager(self.instance)
+            self.cleanup_paths.append(pod_manager.kube_config)
+            try:
+                log_name = self.instance.log_format
+                logger.debug(f"Launching pod for {log_name}.")
+                pod_manager.deploy()
+            except (ApiException, Exception) as exc:
+                if isinstance(exc, ApiException) and exc.status == 403:
+                    try:
+                        if 'exceeded quota' in json.loads(exc.body)['message']:
+                            # If the k8s cluster does not have capacity, we move the job back into
+                            # pending and immediately reschedule the task manager.
+                            logger.warn(exc.body)
+                            logger.warn(f"Could not launch pod for {log_name}. Exceeded quota.")
+                            self.update_model(pk, status='pending')
+                            schedule_task_manager()
```

Comment:
Should we consider sleeping for 5 - 10 seconds here? This is causing us to hit the API pretty hard when there's a lot of jobs trying to run on the cluster.